### ISSUE TITLE
Add V-Dem data

### ISSUE THREAD

**Issue by jhollway** [globalgov/qStates#23]
https://www.v-dem.net/en/data/data/v-dem-dataset-v111/

**Comment by BBieri** [comment]
Hi @jaeltan, I took a look at the V-Dem and V-Party datasets you imported. Good job! From what we discussed and what I have seen, it seems that both datasets are quite big. I have also seen that they have an [R-package](https://github.com/vdeminstitute/vdemdata) with both the updated V-Dem and V-Party datasets that they actively maintain. We could use a similar strategy than for the C-Shapes import and create an interface function that imports, cleans, and converts the data from their package directly. 

The disadvantage of this solution is that we would add a dependency to `{qStates}`. 

What do you think? @jaeltan @henriquesposito

**Comment by jhollway** [comment]
Could it being on GitHub only (for the moment) trigger issues like those we're dealing with qData?

**Comment by BBieri** [comment]
No, this is unrelated to the GitHub actions issues. Basically, the two alternatives I am suggesting are:

- We integrate the raw datasets in our package directly in a CSV file (the traditional workflow)
  - The raw data and the processed data would reside in our package in this case
  - This would make the package quite heavy
- We create a function that retrieves the data from their data package, formats it correctly and loads it into the user's environment (like we did for the C-Shapes data)
  - The raw data and the processed data would not reside in our package but only the `import_VDem()` function
  - This would imply an additional dependency for `qStates`

What do you think of these approaches?

**Comment by jaeltan** [comment]
I think we could give it a try, at the moment the V-Dem dataset can't be pushed onto GitHub at all because of its size, unless we use GitHub Large File Storage (not sure how well this works), or drop some of the data.

**Comment by henriquesposito** [comment]
Thank you @jaeltan and @BBieri for the good work and helpful comments looking for possible solutions.

1. If we decide to add/store the VDem data into qStates, we should perhaps look at different file formats that are lighter than CSV. On top of my head, and from some of my work I have done on text, Rds is much faster and occupy less disk space than CSV. There is also the feather format/package which can be helpful for these types of operation (https://bookdown.org/csgillespie/efficientR/input-output.html). If you are able to bring the data to less than 100 MB, then, GitHub will let you push and just give you a warning. It is true that the package gets heavier, but it really only slows down (noticeably) the package once the data is loaded into environment and starts to use memory (and I think that this is the case whether you download the data or have it stored).

2. Downloading the data as we do for CShapes is indeed a neat solution here. Especially if we expect the datasets to be updated from time to time, that is, we can keep up to date more easily. However, GitHub has an API download limit for "regular" R users. Depending on the size of the files we would download, we would have to pay attention to this issue. For example, `qData::get_packages()` only works a few times in a row or in a single day before exceeding the API rate limit and giving the famous 404 error. 

I believe before making a decision here, we should see (a) whether the file size can be reduced and more efficiently stored, and (b) whether GitHub API limits would be an issue for creating a function for downloading this dataset specifically and, if so, whether there is a solution for it. 

A decision on this issue may also be important for setting precedent for how to proceed when adding large datasets into various qPackages. Thank you.

**Comment by henriquesposito** [comment]
Great job @BBieri, thank you! 

`standardise_titles()` is designed to for agreement titles (sentences) and some of the things that take the most time to process relate to removing all punctuation from these sentences, for examples. These types of operations you are not need for country names or labels. Thus, if you are passing `standardise_titles()` to country names or country labels you can, instead, simply use the text wrangling cleaning/standardizing functions you need more specifically; if they are needed. Otherwise, no need for `standardise_titles()` as I see it.

Once again, great job in getting this done!

**Comment by BBieri** [comment]
Thanks for the quick answer @henriquesposito !

Perfect! I'll remove these calls in that case then if that sounds good to @jaeltan too since she knows more about the data itself. 

Have a good one!
